src/uct/ib/dc/accel/dc_mlx5.c

```diff
@@ -619,9 +622,9 @@ ucs_status_t uct_dc_mlx5_ep_fc_ctrl(uct_ep_t *tl_ep, unsigned op,
                                      txqp, txwq, MLX5_OPCODE_SEND_IMM,
                                      &dc_mlx5_ep->av /*dummy*/, 0, op, sender_ep,
                                      iface->super.rx.dct->dct_num,
-                                     0, 0,
-                                     &dc_mlx5_ep->av,
-                                     uct_ib_mlx5_wqe_av_size(&dc_mlx5_ep->av));
+                                     0, 0, &dc_mlx5_ep->av,
+                                     uct_ib_mlx5_wqe_av_size(&dc_mlx5_ep->av),
+                                     MLX5_WQE_CTRL_SOLICITED);
```

Comment:
Why do you send HARD_REQ with solicited flag? In DC hard request is a separate AM as well as PURE grant. Is the goal to get rid of all auxilliary FC notifications?

Reply:
as we discussed, need to wake up a receiver (which potentially can go to sleep) to send back credit grants, to avoid deadlock.